librz/bin/bfile.c

```diff
@@ -426,43 +426,47 @@ RZ_API RZ_OWN RzPVector /*<RzBinFileHash *>*/ *rz_bin_file_compute_hashes(RzBin
 	if (!md) {
 		goto rz_bin_file_compute_hashes_bad;
 	}
+	RzListIter *hash_algos_iter = bin->default_hashes ? rz_list_iterator(bin->default_hashes) : NULL;
 
-	if (!rz_hash_cfg_configure(md, "md5") ||
-		!rz_hash_cfg_configure(md, "sha1") ||
-		!rz_hash_cfg_configure(md, "sha256") ||
-		!rz_hash_cfg_configure(md, "crc32") ||
-		!rz_hash_cfg_configure(md, "entropy")) {
-		goto rz_bin_file_compute_hashes_bad;
-	}
-	if (!rz_hash_cfg_init(md)) {
-		goto rz_bin_file_compute_hashes_bad;
-	}
+	if (hash_algos_iter) {
```

Comment:
i would change this as, if `hash_algos_iter` is `NULL` then goto `rz_bin_file_compute_hashes_bad`.
Maybe also throw an error via `RZ_LOG_ERROR`